scripts/lib/CIME/test_scheduler.py

```diff
@@ -676,6 +676,13 @@ def _xml_phase(self, test):
                     if match.group(2):
                         envtest.set_test_parameter("PIO_STRIDE_CPL",match.group(2))
 
+                elif opt.startswith('G'):
```

Comment:
I think that you don't want this here.  You've already done this in create_newcase so all you need to do is add G to the elif statement below and at around line 458 add 
--ngpus_per_node to the create_newcase_cmd

Reply:
Thanks @jedwards4b! I just revised this part based on your suggestions and now it works.